HVSample/src/main/java/com/microsoft/hsg/android/hvsample/WeightActivity.java

```diff
@@ -30,112 +37,120 @@
+	}
+
+	@Override
+	protected void onResume() {
+		super.onResume();
+		getWeights();
 	}
-    
-    @Override
-    protected void onResume()
-    {
-        super.onResume();
-        currentRecord = HealthVaultApp.getInstance().getCurrentRecord();
-        getWeights();
-    }
-    
-    @Override
+
+	@Override
 	protected void onStop() {
-    	hvClient.stop();
+		mClient.stop();
 		super.onStop();
 	}
 
-    @SuppressWarnings("unchecked")
-    private void getWeights()
-    {
-    	hvClient.asyncRequest(
-    			currentRecord.getThingsAsync(ThingRequestGroup2.thingTypeQuery(Weight.ThingType)),
-    			new WeightCallback(WeightCallback.RenderWeights));
-    }
-    
-    private void putWeight(String value) {
-        final Thing2 thing = new Thing2();
+	@SuppressWarnings("unchecked")
+	private void getWeights() {
+		mCurrentRecord = HealthVaultApp.getInstance().getCurrentRecord();
+		mClient.asyncRequest(
+				mCurrentRecord.getThingsAsync(ThingRequestGroup2.thingTypeQuery(Weight.ThingType)),
+				new WeightCallback(WeightCallback.RenderWeights));
+	}
+
+	private void putWeight(String value) {
+		final Thing2 thing = new Thing2();
 		thing.setData(new Weight(Double.parseDouble(value)));
-        // thing.setData(new Weight(-10));
-		hvClient.asyncRequest(
-				currentRecord.putThingAsync(thing),
+		mClient.asyncRequest(
+				mCurrentRecord.putThingAsync(thing),
 				new WeightCallback(WeightCallback.PutWeights));
-    }
-    
-    private void renderWeights(List<Thing2> things) {
-    	List<String> weights = new ArrayList<String>();
-        for(Thing2 thing : things) {
-        	Weight w = (Weight)thing.getData();
-        	weights.add(String.valueOf(w.getValue().getKg()));	
-        }
-        ListView lv = (ListView)findViewById(R.id.weightList);
-        lv.setAdapter(new ArrayAdapter<String>(
-            WeightActivity.this,
-            android.R.layout.simple_list_item_1,
-            weights));
-    }
-    
-    public class WeightCallback<Object> implements RequestCallback {
-    	public final static int RenderWeights = 0;
-    	public final static int PutWeights = 1;
-    
-    	private int event;
-    	
-    	public WeightCallback(int event) {
-            WeightActivity.this.setProgressBarIndeterminateVisibility(true);
-    		this.event = event;
-    	}
-
-        @Override
-        public void onError(HVException exception) {
-            WeightActivity.this.setProgressBarIndeterminateVisibility(false);
-            Toast.makeText(
-                WeightActivity.this, 
-                "An error occurred.  " + exception.getMessage(), 
-                Toast.LENGTH_LONG).show();
-        }
+	}
+
+	private void renderWeights(List<Thing2> things) {
+		List<String> weights = new ArrayList<String>();
+		for(Thing2 thing : things) {
+			Weight w = (Weight)thing.getData();
+			weights.add(String.valueOf(w.getValue().getKg()));
+		}
+		ListView lv = (ListView)findViewById(R.id.weightList);
+		lv.setAdapter(new ArrayAdapter<String>(
```

Comment:
As mentioned earlier, we should set the adapter once and the just update it. You have done this elsewhere in this review, so just apply the same logic.